src/havard_eig_ex/ex05/myrand.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+
+__author__ = 'Håvard Brobakken Eig'
+__email__ = 'havardei@nmbu.no'
+
+
+class LCGRand:
+    slope = 7 ** 5
+    congruence_class = 2 ** 31 - 1
+
+    def __init__(self, seed):
+        """
+        Initialise a linear congruence random number generator
+
+        Arguments
+        ---------
+        seed : int
+            The initial seed for the generator
+        """
+        self._hidden_state = seed
+
+    def rand(self):
+        """
+        Generate a single random number.
+
+        Returns
+        -------
+        int
+            A random integer
+        """
+        self._hidden_state *= self.slope
+        self._hidden_state %= self.congruence_class
+
+        return self._hidden_state
+
+    def random_sequence(self, length):
+        return RandIter(self, length)
+
+    def infinite_random_sequence(self):
+        """
+        Generate an infinite sequence of random numbers.
+        Yields
+        ------
+        int
+            A random number.
+        """
+        return RandIter(self, -1)
```

Comment:
It said `yield` in the docstring and you return an iterator 😄 . Not wrong, but what we expected is
```
while True:
   yield self.rand()
```